wandb/file_pusher.py

```diff
@@ -21,14 +29,30 @@
 TMP_DIR = tempfile.TemporaryDirectory('wandb')
 
 
+EventFileBatch = collections.namedtuple(
+    'EventFileBatch', ('batch_id', 'file_changed_events',))
 EventFileChanged = collections.namedtuple(
     'EventFileChanged', ('path', 'save_name', 'copy'))
 EventJobDone = collections.namedtuple('EventJobDone', ('job'))
 EventFinish = collections.namedtuple('EventFinish', ())
 
+# After 15 seconds of gathering batched uploads, kick off a batch without
+# waiting any longer.
+BATCH_THRESHOLD_SECS = 15
```

Comment:
I vote for making this 1 or 2 seconds.

Reasoning:
- If we're using a "sync immediately" policy on a file, it's because we want it be viewable in the UI quickly. A 15 second could be enough to be annoying.
- I would think files coming into filepusher are very bursty. It'd be typical to get them all from a single for loop that finishes in less than a second. (e.g. in the dataframe case they come in at basically the same time). So a 1 second policy would likely have a very similar amount of batching for standard use cases.